full_name.rb

```diff
@@ -0,0 +1,22 @@
+class FullName
+  def initialize()
+    @name_array = []
+    print "What's your first name? "
+    @name_array.push(gets.chomp)
+    print "What's your middle name? "
+    @name_array.push(gets.chomp)
+    print "What's your last name? "
+    @name_array.push(gets.chomp)
+  end
+
+  def greet
+    full_name = ""
+    @name_array.each { |name|
+      full_name += "#{name} "
+    }
```

Comment:
I like how you did this but wanted to make you aware that Ruby has a built in helper method you can call on an array called `.join`, which you could use to do a similar thing to what you are doing here.

`@name_array.join(' ')` would join each item in an array into a string with a space ' ' in between each item.

[Ruby docs](https://ruby-doc.org/core-3.1.1/Array.html#method-i-join)

Reply:
Thank you. Helpful to know.